airflow/contrib/hooks/livy_hook.py

```diff
@@ -0,0 +1,297 @@
+            raise AirflowException("Unable to parse a batch session id")
+        self.log.info("Batch submitted with session id: {}".format(batch_id))
+
+        return batch_id
+
+    def get_batch(self, session_id):
+        """
+        Fetch info about the specified batch
+        :param session_id: identifier of the batch sessions
+        :type session_id: int
+        """
+        LivyHook._validate_session_id(session_id)
+
+        self.log.debug("Fetching info for batch session {}".format(session_id))
+        response = requests.get('{}/batches/{}'.format(self._base_url, session_id))
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Unable to fetch batch with id: {}".format(session_id))
+
+        return response.json()
+
+    def get_batch_state(self, session_id):
+        """
+        Fetch the state of the specified batch
+        :param session_id: identifier of the batch sessions
+        :type session_id: int
+        """
+        LivyHook._validate_session_id(session_id)
+
+        self.log.debug("Fetching info for batch session {}".format(session_id))
+        response = requests.get('{}/batches/{}/state'.format(self._base_url, session_id))
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Unable to fetch state for batch id: {}".format(session_id))
+
+        jresp = response.json()
+        if 'state' not in jresp:
+            raise AirflowException("Unable to get state for batch with id: {}".format(session_id))
+        return BatchState(jresp['state'])
+
+    def delete_batch(self, session_id):
+        """
+        Delete the specified batch
+        :param session_id: identifier of the batch sessions
+        :type session_id: int
+        """
+        LivyHook._validate_session_id(session_id)
+
+        self.log.info("Deleting batch session {}".format(session_id))
+        response = requests.delete('{}/batches/{}'.format(self._base_url, session_id))
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Could not kill the batch with session id: {}".format(session_id))
+
+        return response.json()
+
+    @staticmethod
+    def _validate_session_id(session_id):
+        try:
+            int(session_id)
+        except (TypeError, ValueError):
+            raise AirflowException("'session_id' must represent an integer")
+
+    @staticmethod
+    def _parse_post_response(response):
+        """Parse batch response for batch id"""
+        return response['id'] if 'id' in response else None
+
+    @staticmethod
+    def build_post_batch_body(
+        file,
+        args=None,
+        conf=None,
+        **kwargs
```

Comment:
I would suggest to define the `kwargs` here:
```suggestion
        class_name=None,
        jars=None,
        py_files=None,
        files=None,
        archives=None,
        name=None,
        driver_memory=None,
        driver_cores=None,
        executor_memory= None,
        executor_cores=None,
        num_executors=None,
        queue=None,
        proxy_user=None
```
Then you could remove:
```diff
- class_name = kwargs.get('class_name')
- jars = kwargs.get('jars')
- py_files = kwargs.get('py_files')
- files = kwargs.get('files')
- archives = kwargs.get('archives')
- name = kwargs.get('name')
- driver_memory = kwargs.get('driver_memory')
- driver_cores = kwargs.get('driver_cores')
- executor_memory = kwargs.get('executor_memory')
- executor_cores = kwargs.get('executor_cores')
- num_executors = kwargs.get('num_executors')
- queue = kwargs.get('queue')
- proxy_user = kwargs.get('proxy_user')
```